### ISSUE TITLE
`Style/AccessModifierDeclarations`'s autocorrect detaches method comments

### ISSUE THREAD

**Issue by r7kamura** [rubocop/rubocop#11051]
`Style/AccessModifierDeclarations`'s autocorrect detaches method comments

As title says, for example, in the following code I would expect `Style/AccessModifierDeclarations`'s autocorrect to preserve the method comments, but in fact it does not.

```ruby
# frozen_string_literal: true

class A
  # @param bar [Integer]
  # @return [Integer]
  private def foo(bar)
    bar + 1
  end
end
```

## Expected behavior

```ruby
# frozen_string_literal: true

class A
  private

  # @param bar [Integer]
  # @return [Integer]
  def foo(bar)
    bar + 1
  end
end
```

## Actual behavior

```ruby
# frozen_string_literal: true

class A
  # @param bar [Integer]
  # @return [Integer]

  private

  def foo(bar)
    bar + 1
  end
end
```

## Steps to reproduce the problem

Run `rubocop -A` to the above code with default configuration.

Or we can reproduce it by the following test case:

```diff
diff --git a/spec/rubocop/cop/style/access_modifier_declarations_spec.rb b/spec/rubocop/cop/style/access_modifier_declarations_spec.rb
index b0a78ca11..e501288e7 100644
--- a/spec/rubocop/cop/style/access_modifier_declarations_spec.rb
+++ b/spec/rubocop/cop/style/access_modifier_declarations_spec.rb
@@ -202,6 +202,31 @@ RSpec.describe RuboCop::Cop::Style::AccessModifierDeclarations, :config do
         end
       end
 
+      context 'when method has comments' do
+        it 'registers and autocorrects an offense' do
+          expect_offense(<<~RUBY, access_modifier: access_modifier)
+            class Test
+              # comment
+              #{access_modifier} def foo
+              ^{access_modifier} `#{access_modifier}` should not be inlined in method definitions.
+                # comment
+              end
+            end
+          RUBY
+
+          expect_correction(<<~RUBY)
+            class Test
+            #{access_modifier}
+
+            # comment
+            def foo
+              # comment
+            end
+            end
+          RUBY
+        end
+      end
+
       include_examples 'always accepted', access_modifier
     end
   end
```

## RuboCop version

Oringally I confirmed this issue in my personal project with the following versions:

```
$ bundle exec rubocop -V
1.36.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.1.2) [x86_64-linux]
  - rubocop-performance 1.15.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.13.2
```
